integration-tests/integration_test.go

```diff
@@ -0,0 +1,1021 @@
+			t.Log("Validating response...")
+			validateResponse(t, resultSet)
+		} else {
+			t.Logf("WARNING: Received redemption message for different request ID: %s (expected: %s)",
+				resultSet.Request_id, requestID)
+		}
+
+		require.NotNil(t, resultSet, "Redemption response set should not be nil")
+	}
+
+	t.Log("KAFKA TOKEN ISSUANCE AND REDEMPTION FLOW TEST COMPLETED")
+}
+func TestHTTPIssuerGetEndpoint(t *testing.T) {
+	t.Log("TESTING HTTP ISSUER GET ENDPOINT")
+	issuerName := "TestIssuer-" + testIssuerSuffix.String()
+
+	t.Logf("Retrieving issuer information for '%s'...", issuerName)
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://cbp:2416/v1/issuer/%s", issuerName), nil)
+	require.NoError(t, err, "failed to create GET issuer HTTP request")
+
+	client := &http.Client{Timeout: 30 * time.Second}
+	resp, err := client.Do(req)
+	require.NoError(t, err, "failed to make GET issuer HTTP request")
+	defer resp.Body.Close()
+
+	t.Logf("Issuer GET response status: %s", resp.Status)
+	require.Equal(t, http.StatusOK, resp.StatusCode, "GET issuer request should succeed")
+
+	var issuerResp issuerResponse
+	err = json.NewDecoder(resp.Body).Decode(&issuerResp)
+	require.NoError(t, err, "failed to decode issuer response")
+
+	t.Logf("Issuer details: ID=%s, Name=%s, Cohort=%d",
+		issuerResp.ID, issuerResp.Name, issuerResp.Cohort)
+
+	assert.NotEmpty(t, issuerResp.ID, "issuer ID should not be empty")
+	assert.Equal(t, issuerName, issuerResp.Name, "issuer name should match")
+	assert.NotNil(t, issuerResp.PublicKey, "public key should not be nil")
+	assert.Equal(t, int16(1), issuerResp.Cohort, "cohort should be 1")
+
+	if issuerResp.ExpiresAt != "" {
+		expiresAt, err := time.Parse(time.RFC3339, issuerResp.ExpiresAt)
+		require.NoError(t, err, "failed to parse expires_at timestamp")
+
+		if expiresAt.After(time.Now()) {
+			t.Logf("Expiration is correctly set in the future: %s", expiresAt.Format(time.RFC3339))
+		} else {
+			t.Logf("WARNING: Expiration is in the past: %s", expiresAt.Format(time.RFC3339))
+		}
+
+		assert.True(t, expiresAt.After(time.Now()), "expiration should be in the future")
+	} else {
+		t.Log("Note: Issuer doesn't have an expiration time set")
+	}
+
+	t.Log("Successfully validated issuer response")
+
+	t.Log("Testing retrieval of non-existent issuer...")
+	req, err = http.NewRequest("GET", "http://cbp:2416/v1/issuer/NonExistentIssuer", nil)
+	require.NoError(t, err, "failed to create GET issuer HTTP request for non-existent issuer")
+
+	resp, err = client.Do(req)
+	require.NoError(t, err, "failed to make GET issuer HTTP request for non-existent issuer")
+	defer resp.Body.Close()
+
+	t.Logf("Non-existent issuer GET response status: %s", resp.Status)
+	assert.Equal(t, http.StatusNotFound, resp.StatusCode, "non-existent issuer should return 404")
+	t.Log("Non-existent issuer correctly returned 404 Not Found")
+
+	t.Log("HTTP ISSUER GET ENDPOINT TEST PASSED")
+}
+func TestTokenIssuanceViaKafkaAndRedeemViaHTTPFlow(t *testing.T) {
```

Comment:
suggestion: break test into sub tests and make calls and assertions in the individual test

Its hard follow whats happening in these tests and there is a lot of coupling between issuing tokens and redeeming withing the test suite. For long term maintainability it might be easier to break the code up. For example, 

```
	// create issuer
	
	t.Run("sign_tokens", func(t *testing.T) {
		// ...
	})

	t.Run("map_tokens", func(t *testing.T) {
		// ...
	})

	t.Run("redeem_v3_success", func(t *testing.T) {
		// make http redeem calls
		// assert
	})

	t.Run("redeem_v3_duplicate", func(t *testing.T) {
		// make http redeem calls
		// assert
	})
```